packages/backend/src/core/discovery/proxies/auto/StarkWareProxy.ts

```diff
@@ -169,14 +170,41 @@ async function getStarkWareDiamond(
     data = abi.decodeFunctionData('upgradeTo', tx.data).data as string
   } catch (e) {
     if (
-      lastUpgrade.topics[0] !== coder.getEventTopic('ImplementationUpgraded')
+      !(e instanceof Error) ||
+      !e.message.includes('data signature does not match function upgradeTo.')
     ) {
       throw e
     }
-    console.log(
-      'Failed to decode upgradeTo data, falling back to decoding logs parameters',
-    )
-    data = `0x${lastUpgrade.data.slice(130)}`
+    console.log('Failed to decode upgradeTo data')
+
+    if (lastUpgrade.topics[0] === coder.getEventTopic('Upgraded')) {
```

Comment:
this works for now, but tbh it really is becoming a pain point in our discovery, maybe all starkex deployments share the same storage layout? and we can somehow access the values?

Reply:
I really don't think they do, but that's maybe with checking